app/main.py

```diff
@@ -1,3 +1,63 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(
+                self.km + other.km
+            )
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Distance) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        self.km += other
+        return self
+
+    def __mul__(self, other: Distance) -> Distance:
+        if isinstance(other, Distance):
+            self.km = round(self.km * other)
+            return self
+        return Distance(self.km * other)
```

Comment:
In the `__mul__` method, when `other` is an instance of `Distance`, you should multiply `self.km` by `other.km` instead of `other`. This is an error related to the task requirements.